src/platforms/generic_unix/lib/sys.c

```diff
@@ -98,8 +109,66 @@ struct GenericUnixPlatformData
 #endif
 #endif
 #endif
+
+#ifdef ATOMVM_HAS_MBEDTLS
+#ifndef AVM_NO_SMP
+    Mutex *entropy_mutex;
+#endif
+    mbedtls_entropy_context entropy_ctx;
```

Comment:
We probably want the mbed tls contexts to be in GlobalContext and be available in all platforms unless ATOMVM_HAS_MBEDTLS is undefined.

Reply:
I'd like to do some further decoupling while iterating over, so I don't like turning GlobalContext into a kitchen sink.

So the plan is, as mentioned in #930, shipping some features using common components and libraries, and then features such as mbedtls are picked up from platforms supporting it (+ implementing the missing pieces).